src/libstd/io/buffered.rs

```diff
@@ -321,12 +388,39 @@ impl<W: Write> BufWriter<W> {
     /// # Warning
     ///
     /// It is inadvisable to directly write to the underlying writer.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::io::prelude::*;
+    /// use std::io::BufWriter;
+    /// use std::net::TcpStream;
+    ///
+    /// let mut buffer = BufWriter::new(TcpStream::connect("127.0.0.1:34254").unwrap());
+    ///
+    /// let reference = buffer.get_mut();
+    ///
+    /// // we can use reference just like buffer
+    /// reference.write(&[42]).unwrap();
```

Comment:
This may want to be a slightly different example, the warning above indicates that this should *not* be done (bypassing the buffer), so it seems somewhat odd to turn around and call this.

It may be best to either update the comment here just leave it at `let reference` like the above example.